corehq/apps/locations/templates/locations/manage/locations.html

```diff
@@ -264,4 +310,22 @@
         <button data-bind="click: archive_fn" class="btn btn-warning">{% trans "Archive" %}</button>
     </div>
 </div>
+<div class="modal hide fade" id="delete-location-modal">
+    <div class="modal-header">
+        <a class="close" data-dismiss="modal">&times;</a>
+        <h4>{% trans "Delete Location:" %} <span data-bind="text: name" \></h4>
+    </div>
+    <div class="modal-body">
+        <p>
+            {% blocktrans %}
+            <strong>Warning!</strong> Deleting a location is permanent. There is no way to undo this action.
+            This will also delete all child locations. Are you sure you want to delete this location?
+            {% endblocktrans %}
```

Comment:
It would also be nice to show how many child locations will be deleted.  Something like:
```django
{% if not num_children %}
You are about to delete a location.
{% else %}
You are about to delete this location and its {{ num_children }} child locations.
{% endif %}
Are you sure....
```